app/src/page/splash/view.js

```diff
@@ -76,8 +87,27 @@ class SplashScreen extends React.PureComponent {
       });
       return;
     }
-    if (status.blockchain_status && status.blockchain_status.blocks_behind > 0) {
-      const behind = status.blockchain_status.blocks_behind;
+
+    const blockchainStatus = status.blockchain_status;
+    if (blockchainStatus) {
+      this.setState({
+        isDownloadingHeaders: blockchainStatus.is_downloading_headers,
+        headersDownloadProgress: blockchainStatus.headers_download_progress
+      });
+    }
+
+    if (blockchainStatus && (blockchainStatus.is_downloading_headers ||
+      (this.state.didDownloadHeaders && 'loading_wallet' === startupStatus.code))) {
```

Comment:
`"loading_wallet"` should be a constant with other startup codes

Reply:
This is the only place it's being used at the moment, but I believe we can do this at the point where we intend to refer to the string from other source files.